chainstate/src/lib.rs

```diff
@@ -99,3 +102,35 @@ pub fn make_chainstate<
     let chainstate_interface = ChainstateInterfaceImpl::new(chainstate);
     Ok(Box::new(chainstate_interface))
 }
+
+#[doc(hidden)]
+pub mod integration_tests_support {
+    use super::*;
+
+    pub use crate::interface::chainstate_interface::integration_tests_support::*;
+
+    // This is the same as the normal make_chainstate, but it returns ChainstateTestInterface
+    // instead.
+    pub fn make_chainstate<
```

Comment:
If you're going this far, why not just keep the tests locally here, in chainstate, instead of moving them to the framework?

Reply:
Well, they depend on the framework and in general they are integration tests rather than unit tests.

Reply:
[discussed] I removed the `integration_tests_support` module and the checks of BestChainCandidates that were performed by the integration tests. Then I'll have to refactor BestChainCandidates to abstract away ChainstateRef, so that it can be unit-tested more easily. Since at this moment there are more important things to do, this PR will be postponed.